src/GeositeFramework/js/Map.js

```diff
@@ -206,33 +204,18 @@ require(['use!Geosite',
             id = 'legend-container-' + mapNumber,
             legend = new Legend(regionData, id);
 
-        var getServiceLegend = (function() {
-            var cache = {};
-            // Memoize so only 1 request is generated per service.
-            var fetch = _.memoize(function(url) {
-                xhr(url, {
-                        query: 'f=json',
-                        method: 'GET',
-                        handleAs: 'json',
-                        headers: {
-                            // Fixes: "Request header field X-Requested-With is not allowed by Access-Control-Allow-Headers in preflight response."
-                            'X-Requested-With': null
-                        }
-                    })
-                    .then(function(data) {
-                        cache[url] = data.layers;
-                    })
-                    .then(redraw);
-            });
-            return function(serviceUrl) {
-                var url = serviceUrl + '/legend';
-                if (cache[url]) {
-                    return cache[url];
-                }
-                fetch(url);
-                return null;
-            };
-        }());
+        var redraw = function() {
+            legend.render(getVisibleLayers());
+        };
+
+        function getServiceLegend(service) {
+            var legendUrl = service.url + '/legend',
+                data = ajaxUtil.get(legendUrl);
+            if (ajaxUtil.shouldFetch(legendUrl)) {
```

Comment:
I don't think you can call `fetch` inside `get` because there must be a way to indicate which action to perform after the request completes.

The usefulness of having these 2 distinct operations is more apparent in the layer selector v2 where the usage is a little different. In that case, we only `fetch` when a layer is "activated" (lazy) but we `get` every time the tree is redrawn. Compared to the layer selector, the legend is eager.

Since `get` just returns a cached response, you *could* rewrite this to use `fetch` only, but then the code would be riddled with nested async callbacks and debugging and error handling would become much more complicated.